subprojects/ide/src/integTest/groovy/org/gradle/plugins/ide/eclipse/EclipseTestConfigurationsWithExternalDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugins.ide.eclipse
+
+import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+
+class EclipseTestConfigurationsWithExternalDependenciesIntegrationTest extends AbstractEclipseTestSourcesIntegrationTest {
+
+    def setup() {
+        mavenRepo.module('org', 'lib').publish()
+        buildFile << """
+            plugins {
+                id 'eclipse'
+                id 'java-library'
+            }
+
+            repositories {
+                maven { url "${mavenRepo.uri}" }
+            }
+        """
+    }
+
+    @ToBeFixedForConfigurationCache
+    def "dependencies in main source set dependency configurations are not marked with test classpath attribute"() {
+        given:
+        buildFile << """
+            dependencies {
+                implementation 'org:lib:1.0'
+            }
+        """
+
+        when:
+        run 'eclipse'
+
+        then:
+        assertJarDependencyDoesNotHaveTestAttribute('lib-1.0.jar')
+    }
+
+    @ToBeFixedForConfigurationCache
+    def "dependencies in test source set dependency configurations are marked with test classpath attribute"() {
+        given:
+        buildFile << """
+            dependencies {
+                testImplementation 'org:lib:1.0'
+            }
+        """
+
+        when:
+        run 'eclipse'
+
+        then:
+        assertJarDependencyHasTestAttribute('lib-1.0.jar')
+    }
+
+    @ToBeFixedForConfigurationCache
+    def "dependencies in jvm test suites are marked with test classpath attribute"() {
```

Comment:
I'd add an additional test for a suite which doesn't add an explicit dependency, but uses one of the `useJUnitJupiter()` methods to add its dependencies, just to be sure.

Reply:
Done